src/ServiceControl.Transports.ASQ/ASQTransportCustomization.cs

```diff
@@ -23,6 +24,10 @@ static void ConfigureTransport(TransportExtensions<AzureStorageQueueTransport> t
             transport.SanitizeQueueNamesWith(BackwardsCompatibleQueueNameSanitizer.Sanitize);
             transport.Transactions(TransportTransactionMode.ReceiveOnly);
             transport.ConnectionString(transportSettings.ConnectionString);
+
+            transport.MessageInvisibleTime(TimeSpan.FromMinutes(5));
+            transport.BatchSize(1);
+            transport.DegreeOfReceiveParallelism((int)Math.Sqrt(Math.Min(Environment.ProcessorCount, transportSettings.MaxConcurrency)));
```

Comment:
Is that right? If we have 16 processors then we can only have Receive Parallelism of 4?

Reply:
[Doco](https://docs.particular.net/transports/azure-storage-queues/configuration#configuration-parameters-degreeofreceiveparallelism) says `sqrt(maxConcurrency)`. And concurrency is `max(Number of logical processors, 2)`. Therefore, shouldn't it be `sqrt(settings.Concurrency)` only?

Reply:
If we don't set anything the default is `sqrt(settings.Concurrency)` so I vote for removing this line. Let's stick with whatever transport default it.